### PR TITLE
fix(storybook): harden typography loading

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

- Additional UI styling for docs visibility/usability
- Font loading improvements to signal completion by Typekit

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] Expect to see the Adobe Clean font loaded consistently across [Storybook](https://pr-2930--spectrum-css.netlify.app/preview)
- [x] Expect to see the Adobe Clean font loaded consistently in all visual regression tests
- [ ] Open [Storybook](https://pr-2930--spectrum-css.netlify.app/preview) and open an inspector to see the console log. Expect to see `Font loaded [id: rok6rmo]` on first page load.
- [x] Open [Storybook](https://pr-2930--spectrum-css.netlify.app/preview) and open an inspector to see the console log. Change the language switcher to `Japanese`. Expect to see `Font loaded [id: mge7bvf]` in the log.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: .storybook/loaders/font-loader.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const FontLoader = async () => ({
```
Comment (jawinn) [inline .storybook/loaders/font-loader.js:1]: Where is this being loaded / imported?

**Review by jawinn — CHANGES_REQUESTED** [review]
I see there are no VRT changes; is that correct or were there expected changes/fixes for typography in any stories? I see Adobe Clean continuing to load for stories and in VRTs as before.

**Review by pfulton — APPROVED** [review]

File: .storybook/preview-head.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Observe changes to the lang attribute on the html element
+    observer.observe(document.documentElement, { attributes: true });
+</script>
+
+<script type="module">
+    if (!("anchorName" in document.documentElement.style)) {
+      import("https://unpkg.com/@oddbird/css-anchor-positioning");
+    }
+  </script>
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply .storybook/preview-head.html:42]: Yes thank you, should not have gotten in there!

File: .storybook/preview-head.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                const lang = mutation.target.getAttribute("lang");
+                const isNotEnglish = lang && lang !== "en-US";
+                const kitId = isNotEnglish ? "mge7bvf" : "rok6rmo";
+
+                if (window.Typekit) {
+                    window.Typekit.load({ ...config, kitId });
+                }
+            }
+        });
+    });
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply .storybook/preview-head.html:32]: You'll see the console log update when you change the language switcher.  I'll add a test case.

File: .storybook/decorators/language.js [hunk 4]
```diff
@@ -17,6 +18,34 @@ export const withLanguageWrapper = makeDecorator({
...
+				// https://github.com/typekit/webfontloader?tab=readme-ov-file#configuration
+				loading: function() {},
+				fontactive: function(familyName) {
+					console.log(`Font ${familyName} active`);
+				},
+				fontinactive: function(familyName) {
+					console.log(`Font ${familyName} inactive`);
+				},
+				active: function() {
+					console.log(`Font loaded [id: ${kitId}]`);
```
Comment (jawinn) [inline .storybook/decorators/language.js:39]:
```suggestion
					console.log(`Font kit loaded [id: ${kitId}]`);
```

File: .storybook/decorators/language.js [hunk 5]
```diff
@@ -17,6 +18,34 @@ export const withLanguageWrapper = makeDecorator({
...
+			// otherwise use the mge7bvf kit with all the language settings (larger size)
+			const kitId = isNotEnglish ? "mge7bvf" : "rok6rmo";
+			const config = {
+				kitId,
+				async: true,
+				scriptTimeout: 3000,
+				// https://github.com/typekit/webfontloader?tab=readme-ov-file#configuration
+				loading: function() {},
+				fontactive: function(familyName) {
+					console.log(`Font ${familyName} active`);
```
Comment (jawinn) [inline .storybook/decorators/language.js:33]: Should we remove this logging before merge? It's creating a lot of log entries. And duplicative ones for some reason:
Screenshot 2024-07-29 at 10 11 36 AM

**Review by jawinn — APPROVED** [review]
I looked at the verification instructions and was able to see the font kit logging on component pages. And the change of kit when switching to Japanese. 👍 

It doesn't log on the Contributing page for some reason. And the docs page will log it multiple times per story, which I'm guessing is normal? If those aren't issues, I think this should be good to go. And I just had one question about adjusting the font family logging before merge.

File: .storybook/decorators/language.js (hunk 5 above)
Comment (castastrophe) [reply .storybook/decorators/language.js:33]: I do want to go back and look into why it's logging multiple times, let me add a ticket for this.

**Comment by castastrophe** [comment]
> It doesn't log on the Contributing page for some reason. And the docs page will log it multiple times per story, which I'm guessing is normal? If those aren't issues, I think this should be good to go. And I just had one question about adjusting the font family logging before merge.

I noticed that too and I'm not loving it but I don't think it's a blocker to getting this improvement in for the S2 work.  I'm going to add a backlog ticket to return to this and harden it further to ensure it only loads once.

File: .storybook/decorators/language.js (hunk 5 above)
Comment (castastrophe) [reply .storybook/decorators/language.js:33]: Opened in Jira as CSS-866

**Pull request merged**